rgsync/Connectors/redis_connector.py

```diff
@@ -0,0 +1,139 @@
+from rgsync.common import *
+from redisgears import getMyHashTag as hashtag
+
+# redis connection class
+class RedisConnection():
+    def __init__(self, host, port, password=None):
+        self._host = host
+        self._port = port
+        self._password = password
+
+    @property
+    def host(self):
+        return self._host() if callable(self._host) else self._host
+
+    @property
+    def port(self):
+        return self._port() if callable(self._port) else self._port
+
+    @property
+    def password(self):
+        return self._password() if callable(self._password) else self._password
+
+    def Connect(self):
+        from redis.client import Redis
+        WriteBehindLog("Connect: connecting to %s:%d password: %s" % (self.host, self.port, self.password))
+        try:
+            r = Redis(host=self.host, port=self.port, password=self.password)
+        except Exception as e:
+            WriteBehindLog("Cannot connect to Redis. Aborting (%s)" % str(e))
+        WriteBehindLog("Connect: connected")
+        return r
+
+SIMPLE_HASH_BACKEND_PK = 'HashBackendPK'
+SIMPLE_HASH_BACKEND_TABLE = 'HashBackendTable'
+
+# redis connector class
+class RedisConnector():
+    def __init__(self, connection, newPrefix, exactlyOnceTableName=None):
+        self.connection = connection
+        self.session = None
+        self.new_prefix = newPrefix
+        self.supportedOperations = [OPERATION_DEL_REPLICATE, OPERATION_UPDATE_REPLICATE]
+        self.exactlyOnceTableName = exactlyOnceTableName
+        self.exactlyOnceLastId = None
+        self.shouldCompareId = True if self.exactlyOnceTableName is not None else False
+
+    def TableName(self):
+        return SIMPLE_HASH_BACKEND_TABLE
+
+    def PrimaryKey(self):
+        return SIMPLE_HASH_BACKEND_PK
+
+    def WriteData(self, data):
+        # return if no data is received to write
+        if(0 == len(data)):
+            WriteBehindLog("Warning: in connector received empty batch to write")
+            return
+
+        try:
+            if not self.session:
+                self.session = self.connection.Connect()
+                if self.exactlyOnceTableName is not None:
+                    shardId = 'shard-%s' % hashtag()
+                    res = self.session.execute_command('HGET', self.exactlyOnceTableName, shardId)
```

Comment:
Does the redis python client rais exception if it gets an error?

Reply:
As I can check in my installed Redis python module version '3.5.3'.
It raises exception. For reference:

def execute_command(self, *args, **options):
	"Execute a command and return a parsed response"
	pool = self.connection_pool
	command_name = args[0]
	conn = self.connection or pool.get_connection(command_name, **options)
	try:
		conn.send_command(*args)
		return self.parse_response(conn, command_name, **options)
	except (ConnectionError, TimeoutError) as e:
		conn.disconnect()
		if not (conn.retry_on_timeout and isinstance(e, TimeoutError)):
			**raise**
		conn.send_command(*args)
		return self.parse_response(conn, command_name, **options)

Also I have only checked interface for connect, execute command and close connection. I have not explored Redis python client in depth for connection pool and parser class.

Reply:
I think it raises an exception if it sees an error reply:
```
>>> import redis
>>> r = redis.Redis()
>>> r.execute_command('hget')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/redis/client.py", line 755, in execute_command
    return self.parse_response(connection, command_name, **options)
  File "/usr/local/lib/python2.7/dist-packages/redis/client.py", line 768, in parse_response
    response = connection.read_response()
  File "/usr/local/lib/python2.7/dist-packages/redis/connection.py", line 638, in read_response
    raise response
redis.exceptions.ResponseError: wrong number of arguments for 'hget' command
```
So we need to catch the exception here...